lib/data-types.js

```diff
@@ -526,6 +526,36 @@ ARRAY.is = function(obj, type) {
   return obj instanceof ARRAY && obj.type instanceof type;
 };
 
+var GEOMETRY = function(type, srid) {
+  var options = _.isPlainObject(type) ? type : {
+    type: type,
+    srid: srid
+  };
+
+  if (!(this instanceof GEOMETRY)) return new GEOMETRY(options);
+
+  this.options = options;
+  this.type = options.type;
+  this.srid = options.srid;
+};
+util.inherits(GEOMETRY, ABSTRACT);
+
+GEOMETRY.prototype.key = GEOMETRY.key = 'GEOMETRY';
```

Comment:
That's also not a bad idea, but one problem of that could be that the data-types files becomes very long, especially for PG which has a fair amount of specific parsing code (hstore, range)

Letting the datatypes parse values is very close to our idea of the [Datatypes up/down system](https://github.com/sequelize/sequelize/issues/2042) (not implemented yet), so you're on the right track, just ahead of your time :P